### PR TITLE
Login and lazy views

### PR DESCRIPTION
**Pull request by bennypowers**
This PR adds polymerfire elements, login with google, fb, and github, and lazy loaded views.

It also adds some auxiliary tools like eslint

### CODE REVIEW

**Comment by bennypowers** [comment]
Keep in mind the views are just stubs. We will pretty them up later :D

File: src/matching-game-login/matching-game-login.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          user: {
+            type: Object,
+            value: null,
+          },
+
+        };
+      }
+
+      dispatchLogin(event) {
+        this.dispatchEvent(new CustomEvent('login-google', {
```
Comment (ronen-e) [inline src/matching-game-login/matching-game-login.html]: this event relates to any login right (with `google`, `facebook`, `github`) ?
so the name `login-google` is too specific. 
why not make it something like `social-login` or just `login`

Comment (bennypowers) [reply src/matching-game-login/matching-game-login.html]: Good catch. Holdover from an early draft. I'll change it to `firebase-signin`

Comment (bennypowers) [reply src/matching-game-login/matching-game-login.html]: Please see latest diff

**Pull request merged**